lib/modem_key_mgmt/modem_key_mgmt.c

```diff
@@ -194,6 +194,43 @@ int modem_key_mgmt_read(nrf_sec_tag_t sec_tag,
 	return 0;
 }
 
+int modem_key_mgmt_cmp(nrf_sec_tag_t sec_tag,
+		       enum modem_key_mgnt_cred_type cred_type,
+		       const void *buf, u16_t len)
+{
+	int err;
+	size_t size;
+	struct at_param_list cmng;
+
+	if (buf == NULL) {
+		return -EINVAL;
+	}
+
+	err = key_fetch(sec_tag, cred_type);
+	if (err) {
+		return err;
+	}
+
+	at_params_list_init(&cmng, AT_CMNG_PARAMS_COUNT);
+	at_parser_params_from_str(scratch_buf, NULL, &cmng);
+
+	at_params_size_get(&cmng, AT_CMNG_CONTENT_INDEX, &size);
+	if (size != len) {
+		LOG_DBG("Credential length %d bytes (expected %d)", size, len);
+		at_params_list_free(&cmng);
+		return 1;
+	}
+
+	at_params_string_get(&cmng, AT_CMNG_CONTENT_INDEX, scratch_buf, &size);
+	at_params_list_free(&cmng);
+
+	if (memcmp(buf, scratch_buf, len)) {
+		LOG_DBG("Credential data mismatch");
+		at_params_list_free(&cmng);
```

Comment:
`cmng` was already freed, same below.

Reply:
I had made a mess here, should be fine now.